Crashes when resolving a merge conflict and changing the file in the `Files` panel

**Describe the bug**
Crashes when resolving a merge conflict and changing the file in the `Files` panel.

**To Reproduce**
Steps to reproduce the behavior:
1. Merge a branch with merge conflicts
2. Change selected file with conflicts with the `arrow-down` key
3. Lazygit crashes and prints error message

**Expected behavior**
The program not crashing.

**Desktop (please complete the following information):**
 - OS: MacOS 11.0 (with Apple Silicon cpu if that's important)
 - Lazygit Version v0.28.1

**Additional context**
The error message:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x104bd794c]

goroutine 1 [running]:
github.com/jesseduffield/lazygit/pkg/gui/mergeconflicts.findConflicts(0x1400069e800, 0x15ad, 0x1400021c1c0, 0x104e92318, 0x14000212fa0)
	/home/runner/work/lazygit/lazygit/pkg/gui/mergeconflicts/state.go:105 +0x10c
github.com/jesseduffield/lazygit/pkg/gui/mergeconflicts.(*State).SetConflictsFromCat(0x140002dc100, 0x1400069e800, 0x15ad)
	/home/runner/work/lazygit/lazygit/pkg/gui/mergeconflicts/state.go:88 +0x30
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).refreshMergePanel(0x1400021c1c0, 0x0, 0x0)
	/home/runner/work/lazygit/lazygit/pkg/gui/merge_panel.go:163 +0x110
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).handlePickHunk.func1(0x14000136060, 0x104e71a58)
	/home/runner/work/lazygit/lazygit/pkg/gui/merge_panel.go:94 +0x9c
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).withMergeConflictLock(0x1400021c1c0, 0x14000171698, 0x0, 0x0)
	/home/runner/work/lazygit/lazygit/pkg/gui/merge_panel.go:312 +0xa8
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).handlePickHunk(0x1400021c1c0, 0x10475fe9c, 0x0)
	/home/runner/work/lazygit/lazygit/pkg/gui/merge_panel.go:77 +0x44
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).wrappedHandler.func1(0x14000203380, 0x140000bcb40, 0x0, 0x1400008d880)
	/home/runner/work/lazygit/lazygit/pkg/gui/confirmation_panel.go:314 +0x28
github.com/jesseduffield/gocui.(*Gui).execKeybinding(0x14000203380, 0x140000bcb40, 0x1400051d280, 0x2, 0x1, 0x1)
	/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1167 +0x5c
github.com/jesseduffield/gocui.(*Gui).execKeybindings(0x14000203380, 0x140000bcb40, 0x140001718c8, 0x14000586160, 0x14000000180, 0x140001718b8)
	/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1135 +0x1a8
github.com/jesseduffield/gocui.(*Gui).onKey(0x14000203380, 0x140001718c8, 0x14000090030, 0x140000c2000)
	/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1060 +0x280
github.com/jesseduffield/gocui.(*Gui).handleEvent(0x14000203380, 0x140001718c8, 0x0, 0x0)
	/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:651 +0x44
github.com/jesseduffield/gocui.(*Gui).MainLoop(0x14000203380, 0x14000000004, 0x1400031f9d8)
	/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:611 +0x138
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).Run(0x1400021c1c0, 0x0, 0x0)
	/home/runner/work/lazygit/lazygit/pkg/gui/gui.go:540 +0x43c
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError.func1(0x1400021c101, 0x14000030540)
	/home/runner/work/lazygit/lazygit/pkg/gui/gui.go:548 +0x2c
github.com/jesseduffield/lazygit/pkg/utils.SafeWithError(0x1400031fb28, 0x0, 0x0)
	/home/runner/work/lazygit/lazygit/pkg/utils/utils.go:380 +0x5c
github.com/jesseduffield/lazygit/pkg/gui.(*Gui).RunAndHandleError(0x1400021c1c0, 0x31, 0x104c5ba00)
	/home/runner/work/lazygit/lazygit/pkg/gui/gui.go:547 +0x70
github.com/jesseduffield/lazygit/pkg/app.(*App).Run(0x1400029db20, 0x1400027e630, 0x0)
	/home/runner/work/lazygit/lazygit/pkg/app/app.go:242 +0x84
main.main()
	/home/runner/work/lazygit/lazygit/main.go:127 +0x9fc
```

--- a/pkg/gui/mergeconflicts/state.go
+++ b/pkg/gui/mergeconflicts/state.go
@@ -104,6 +104,10 @@ func findConflicts(content string) []*mergeConflict {
 		case "=======":
 			newConflict.middle = i
 		default:
+			// Sometimes these lines look like "<<<<<<< HEAD:foo/bar/baz.go" so handle that case as well.
+			if strings.HasPrefix(trimmedLine, "<<<<<<< HEAD:") {
+				newConflict = &mergeConflict{start: i}
+			}
 			if strings.HasPrefix(trimmedLine, ">>>>>>> ") {
 				newConflict.end = i
 				conflicts = append(conflicts, newConflict)

--- a/pkg/gui/mergeconflicts/state_test.go
+++ b/pkg/gui/mergeconflicts/state_test.go
@@ -1,0 +1,90 @@
+package mergeconflicts
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFindConflicts(t *testing.T) {
+	type scenario struct {
+		name     string
+		content  string
+		expected []*mergeConflict
+	}
+
+	scenarios := []scenario{
+		{
+			name:     "empty",
+			content:  "",
+			expected: []*mergeConflict{},
+		},
+		{
+			name: "various conflicts",
+			content: `++<<<<<<< HEAD
+foo
+++=======
+bar
+++>>>>>>> branch
+
+<<<<<<< HEAD: foo/bar/baz.go
+foo
+bar
+=======
+baz
+>>>>>>> branch
+
+++<<<<<<< MERGE_HEAD
+foo
+++=======
+bar
+++>>>>>>> branch
+
+++<<<<<<< Updated upstream
+foo
+++=======
+bar
+++>>>>>>> branch
+
+++<<<<<<< ours
+foo
+++=======
+bar
+++>>>>>>> branch
+`,
+			expected: []*mergeConflict{
+				{
+					start:  0,
+					middle: 2,
+					end:    4,
+				},
+				{
+					start:  6,
+					middle: 9,
+					end:    11,
+				},
+				{
+					start:  13,
+					middle: 15,
+					end:    17,
+				},
+				{
+					start:  19,
+					middle: 21,
+					end:    23,
+				},
+				{
+					start:  25,
+					middle: 27,
+					end:    29,
+				},
+			},
+		},
+	}
+
+	for _, s := range scenarios {
+		t.Run(s.name, func(t *testing.T) {
+			assert.EqualValues(t, s.expected, findConflicts(s.content))
+		})
+	}
+}

